src/pipx/main.py

```diff
@@ -840,7 +840,7 @@ def setup(args: argparse.Namespace) -> None:
         print_version()
         sys.exit(0)
 
-    verbose = args.verbose - args.quiet
+    verbose = getattr(args, "verbose", 0) - getattr(args, "quiet", 0)
```

Comment:
Or we could set argparses' [`default`](https://docs.python.org/3/library/argparse.html#default) keyword argument?

Reply:
It's set to 0 already: https://github.com/pypa/pipx/blob/98635d27d93aff9f74bfca660ae81115d3d37d86/src/pipx/main.py#L724

Reply:
Then I don't understand why this change is needed.

Reply:
See #1177

Reply:
Yes, but `argparse` is supposed to fill in that argument with the default value, isn't it?

I think I've found the culprit: We use a shared parser for defining the verbose and quiet arguments, but don't apply that as a parent parser for the base parser. Thus, the following patch should work:

```patch
diff --git a/src/pipx/main.py b/src/pipx/main.py
index 11e01e5..7dc5d75 100644
--- a/src/pipx/main.py
+++ b/src/pipx/main.py
@@ -720,12 +720,6 @@ def get_command_parser() -> argparse.ArgumentParser:
 
     completer_venvs = InstalledVenvsCompleter(venv_container)
 
-    parser = argparse.ArgumentParser(
-        prog=prog_name(),
-        formatter_class=LineWrapRawTextHelpFormatter,
-        description=PIPX_DESCRIPTION,
-    )
-    parser.man_short_description = PIPX_DESCRIPTION.splitlines()[1]  # type: ignore
 
     shared_parser = argparse.ArgumentParser(add_help=False)
 
@@ -744,6 +738,14 @@ def get_command_parser() -> argparse.ArgumentParser:
         "--verbose", "-v", action="count", default=0, help=("Give more output.")
     )
 
+    parser = argparse.ArgumentParser(
+        prog=prog_name(),
+        formatter_class=LineWrapRawTextHelpFormatter,
+        description=PIPX_DESCRIPTION,
+        parents=[shared_parser],
+    )
+    parser.man_short_description = PIPX_DESCRIPTION.splitlines()[1]  # type: ignore
+
     subparsers = parser.add_subparsers(
         dest="command", description="Get help for commands with pipx COMMAND --help"
     )
```

Reply:
Done